metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/indexbuilder/ESIndexBuilder.java

```diff
@@ -321,6 +322,149 @@ public void buildIndex(ReindexConfig indexState) throws IOException {
     }
   }
 
+  /**
+   * Check if a specific index has 0 replicas and >0 documents, then increase its replica count to
+   * 1.
+   *
+   * @param indexName The name of the index to check
+   * @param dryRun If true, report what would happen without making changes
+   * @return Map containing operation details
+   * @throws IOException if there's an error communicating with Elasticsearch
+   */
+  private Map<String, Object> increaseReplicasForActiveIndices(String indexName, boolean dryRun)
+      throws IOException {
+    Map<String, Object> result = new HashMap<>();
+    result.put("indexName", indexName);
+    result.put("changed", false);
+    result.put("dryRun", dryRun);
+    GetIndexRequest getIndexRequest = new GetIndexRequest(indexName);
+    GetIndexResponse response =
+        _searchClient.indices().get(getIndexRequest, RequestOptions.DEFAULT);
+    Map<String, Settings> indexToSettings = response.getSettings();
+    Settings indexSettings = indexToSettings.get(indexName);
+    int replicaCount = Integer.parseInt(indexSettings.get("index.number_of_replicas", "1"));
```

Comment:
This shouldn't be hard coded. There is the default value in the application properties and then overrides per index to consider. Yes, it is probably always going to be 1, however we expose configuration for this value and should use it.

Reply:
I'm not sure that this has been addressed, this still looks hardcoded to be 1? I think what David is saying is that it should pull from the application.yaml, I.E. IndexConfiguration.numReplicas

Reply:
Nvm previous comment, it's just kind of written strangely. Instead of actually using this replicaCount variable you only are using it to determine if the setting is zero and defaulting to 1 to allow it to go into the block where you use the getter to get the configured value.